pkg/objstore/s3/s3.go

```diff
@@ -213,7 +214,12 @@ func NewBucketWithConfig(logger log.Logger, config Config, component string) (*B
 	if err := validate(config); err != nil {
 		return nil, err
 	}
-	if config.AccessKey != "" {
+
+	if config.AWSSDKAuth {
+		chain = []credentials.Provider{
+			wrapCredentialsProvider(&AWSSDKAuth{Region: config.Region}),
+		}
+	} else if config.AccessKey != "" {
```

Comment:
Can we return error if both accessKey and AWSSDKAuth is set?

Reply:
Sure